### ISSUE TITLE
NetCoreApp2.1 using SDK 2.1.1 package downgrade to 2.1.0

### ISSUE THREAD

**Issue by grantwar** [dotnet/core#1754]
# Service Fabric Service using SDK 2.1.1 has an associated test core 2.1 project which is showing up as using SDK 2.1.0, won't allow me to upgrade via nuget (Blocked by project). 
# Build Error

> Detected package downgrade: Microsoft.NETCore.App from 2.1.1 to 2.1.0. Reference the package directly from the project to select a different version.

# Issue
I'm unable to get my solution to build due to an issue with a unit test project which references my service project forcing a package downgrade of my service project from 2.1.1 down to 2.1.0
No where have I explicitly set the dotnet core revision, and my netcoreapp projects are all 2.1
Prior to me installing dotnetcore 2.1.301, I had the 2.1.3 (RC1) and everything worked just fine.
I'm unable to manually change the version of the unit test project from 2.1.0 to 2.1.1, and doing doing a  install package microsoft.netcore.app -version 2.1.1 does not fix the issue either.

I can set `<TargetLatestRuntimePatch>false</TargetLatestRuntimePatch>` in the service project, and this does resolve the problem, however I'm not happy having to do this in a production service which is being worked on by a team of developers.
# Setup
dotnet --version 
> 2.1.301

Service fabric service project
![image](https://user-images.githubusercontent.com/8554565/42284016-1e4c1910-7fa3-11e8-8dda-216f6d6206c2.png)

Service unit test project
![image](https://user-images.githubusercontent.com/8554565/42284129-788ac034-7fa3-11e8-9597-a67ab7b16712.png)

Service unit test project nuget package manager
![image](https://user-images.githubusercontent.com/8554565/42284200-b0b93ad0-7fa3-11e8-9c08-51a8caee2dbd.png)

**Comment by agr** [comment]
Seeing the same in my project with xUnit when specifying runtime when building. Repro:

```
> dotnet --version
2.1.301
```

```
mkdir 211_210
cd 211_210
mkdir mvc
cd mvc
dotnet new mvc
cd ..
mkdir test
cd test
dotnet new xunit
dotnet add reference ../mvc/mvc.csproj
dotnet build -r any
```

Produces the following output:

```
Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for C:\Users\agr\Source\Repos\211_210\test\test.csproj...
  Restoring packages for C:\Users\agre\Source\Repos\211_210\mvc\mvc.csproj...
  Restore completed in 46.55 ms for C:\Users\agr\Source\Repos\211_210\test\test.csproj.
C:\Users\agr\Source\Repos\211_210\test\test.csproj : error NU1605: Detected package downgrade: Microsoft.NETCore.App from 2.1.1 to 2.1.0. Reference the package directly from the project to select a different version.
C:\Users\agr\Source\Repos\211_210\test\test.csproj : error NU1605:  test -> mvc -> Microsoft.NETCore.App (>= 2.1.1)
C:\Users\agr\Source\Repos\211_210\test\test.csproj : error NU1605:  test -> Microsoft.NETCore.App (>= 2.1.0)
  Restore failed in 294.47 ms for C:\Users\agr\Source\Repos\211_210\test\test.csproj.
  Restore completed in 629.54 ms for C:\Users\agr\Source\Repos\211_210\mvc\mvc.csproj.

Build FAILED.

C:\Users\agr\Source\Repos\211_210\test\test.csproj : error NU1605: Detected package downgrade: Microsoft.NETCore.App from 2.1.1 to 2.1.0. Reference the package directly from the project to select a different version.
C:\Users\agr\Source\Repos\211_210\test\test.csproj : error NU1605:  test -> mvc -> Microsoft.NETCore.App (>= 2.1.1)
C:\Users\agr\Source\Repos\211_210\test\test.csproj : error NU1605:  test -> Microsoft.NETCore.App (>= 2.1.0)
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.03
```

Just `dotnet build` seems to work, but if any runtime is specified it fails. I tried `any`, `win`, `win-x64`, `linux-x64`. Same happens both in Linux and Windows. It worked fine in `2.1.300`.

**Comment by rrelyea** [comment]
This probably requires a fix in the templates or the SDK? (SDK team?)

MVC wants 2.1.1 of NetCoreApp
Unit test projects wants 2.1.0 of NetCoreApp.

Normally, this is a NuGet warning, but SDK projects have turned NU1605 into an error on purpose (Warning as Error).

**Comment by rrelyea** [comment]
My guess is that this should be closed, in favor of the following info I got from @dsplaisted in email:

This is part of a class of issues we have.  If a RID is specified, and the project is an Exe, then by default we’ll use the latest patch of Microsoft.NETCore.App.  We run into issues when you reference a project that uses the latest patch from one that doesn’t.

Other related issues:
https://github.com/dotnet/sdk/issues/2364
https://github.com/dotnet/sdk/issues/2253

**Comment by Petermarcu** [comment]
Issue moved to [dotnet/cli #9624](https://github.com/dotnet/cli/issues/9624) via [**ZenHub**](https://www.zenhub.com/)

**Comment by replaysMike** [comment]
I too am frustrated by this issue.

**Comment by haydenhancock** [comment]
I am also experiencing this issue in VSTS. This is preventing our builds from completing and as others have said, is very frustrating.

**Comment by karelz** [comment]
@Cloudmersive the issue was NOT closed as Won't Fix -- it was MOVED to another repo: https://github.com/dotnet/cli/issues/9624 (see https://github.com/dotnet/core/issues/1754#issuecomment-403620515)
@haydenhancock I would suggest to check solutions & workarounds in the CLI repo issue (or add comments there if needed).

**Comment by haydenhancock** [comment]
@karelz I've tried all the suggested fixes and nothing worked for me. What's interesting is that my build pipeline (with no code changes or build pipeline changes) was working without issue yesterday.

**Comment by karelz** [comment]
@haydenhancock those are good things to raise on the open CLI issue. This issue is closed and further discussion here is not reaching the right experts.
